pkg/ironic/raid.go

```diff
@@ -175,12 +175,20 @@ func buildTargetHardwareRAIDCfg(volumes []metal3v1alpha1.HardwareRAIDVolume) (lo
 			}
 			nameCheckFlags[volume.Name] = index
 		}
+		// Build physicalDisks for logicalDisk
+		physicalDisks := make([]interface{}, len(volume.PhysicalDisks))
+		for i, volumePhysicalDisk := range volume.PhysicalDisks {
+			physicalDisks[i] = volumePhysicalDisk
+		}
 		// Build logicalDisk
 		logicalDisk = nodes.LogicalDisk{
-			SizeGB:     volume.SizeGibibytes,
-			RAIDLevel:  nodes.RAIDLevel(volume.Level),
-			VolumeName: volume.Name,
+			SizeGB:        volume.SizeGibibytes,
+			RAIDLevel:     nodes.RAIDLevel(volume.Level),
+			VolumeName:    volume.Name,
+			Controller:    volume.Controller,
+			PhysicalDisks: physicalDisks,
```

Comment:
So we're just leaving any conflict between NumberOfPhysicalDisks and the list of PhysicalDisks for ironic to resolve?

Reply:
I am trying to validate this in the webhook, but it is creating some problems. I will validate it in some other place in the BMO, if that's alright